File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    void generate(vector<vector<int>> &result,vector<int>&current,int index,int n,vector<int>& nums)
    {
        if(index==n)
        {
            result.push_back(current);
            return;
        }
        generate(result,current,index+1,n,nums);
        current.push_back(nums[index]);
        generate(result,current,index+1,n,nums);
        current.pop_back();
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> result;
        vector<int> current;
        generate(result,current,0,nums.size(),nums);
        return result;
    }
};

File: 0078-subsets/0078-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    const result = [];
    const current = [];
    const generate = (index) => {
        if (index === nums.length) {
            result.push([...current]); 
            return;
        }
        generate(index + 1);
        current.push(nums[index]);
        generate(index + 1);
        current.pop();
    };
    generate(0);
    return result;
};